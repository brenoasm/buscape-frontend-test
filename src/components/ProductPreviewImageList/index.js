import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { colors } from "../../theme";

import ProductPreviewImageItem from "../ProductPreviewImageItem";

const propTypes = {};

const defaultProps = {};

const StyledProductPreviewImageList = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;

  > div:first-child {
    display: flex;
    flex-direction: column;
  }

  > div:last-child {
    > img {
      height: 200px;
    }
  }
`;

const ProductPreviewImageList = ({
  images,
  productId,
  selectedImage,
  onBrokenImageUrl,
  onSelectedImageChange
}) => {
  return (
    <StyledProductPreviewImageList>
      <div>
        {images &&
          images.map(image => (
            <ProductPreviewImageItem
              key={image.url}
              productId={productId}
              url={image.url}
              active={image.selected}
              onBrokenImageUrl={onBrokenImageUrl}
              onSelectedImageChange={onSelectedImageChange}
            />
          ))}
      </div>
      <div>
        <img
          onError={event => onBrokenImageUrl(event)}
          src={selectedImage}
          alt="Imagem do Produto"
        />
      </div>
    </StyledProductPreviewImageList>
  );
};

ProductPreviewImageList.propTypes = propTypes;
ProductPreviewImageList.defaultProps = defaultProps;

export default ProductPreviewImageList;
