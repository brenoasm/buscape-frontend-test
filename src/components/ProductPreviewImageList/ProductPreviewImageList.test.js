import React from "react";
import { shallow } from "enzyme";

import ProductPreviewImageList from "./";

describe("ProductPreviewImageList", () => {
  const initialProps = {
    productId: 1,
    images: [
      {
        url:
          "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
        selected: true
      },
      {
        url:
          "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
        selected: false
      },
      {
        url:
          "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
        selected: false
      },
      {
        url:
          "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
        selected: false
      }
    ],
    selectedImage: "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
  }

  it("matches the snapshot", () => {
    const wrapper = shallow(<ProductPreviewImageList {...initialProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
