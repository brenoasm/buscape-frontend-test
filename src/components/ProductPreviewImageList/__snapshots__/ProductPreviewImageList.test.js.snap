// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPreviewImageList matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductPreviewImageList
    images={
      Array [
        Object {
          "selected": true,
          "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
        },
        Object {
          "selected": false,
          "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
        },
        Object {
          "selected": false,
          "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
        },
        Object {
          "selected": false,
          "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
        },
      ]
    }
    onBrokenImageUrl={[Function]}
    onSelectedImageChange={[Function]}
    productId={1}
    selectedImage="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div>
          <ProductPreviewImageItem
            active={true}
            onBrokenImageUrl={[Function]}
            onSelectedImageChange={[Function]}
            productId={1}
            url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
          />
          <ProductPreviewImageItem
            active={false}
            onBrokenImageUrl={[Function]}
            onSelectedImageChange={[Function]}
            productId={1}
            url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274"
          />
          <ProductPreviewImageItem
            active={false}
            onBrokenImageUrl={[Function]}
            onSelectedImageChange={[Function]}
            productId={1}
            url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138"
          />
          <ProductPreviewImageItem
            active={false}
            onBrokenImageUrl={[Function]}
            onSelectedImageChange={[Function]}
            productId={1}
            url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730"
          />
        </div>,
        <div>
          <Image
            alt="Produto"
            height={200}
            onBrokenImageUrl={[Function]}
            src="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
            width={200}
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ProductPreviewImageItem
              active={true}
              onBrokenImageUrl={[Function]}
              onSelectedImageChange={[Function]}
              productId={1}
              url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
            />,
            <ProductPreviewImageItem
              active={false}
              onBrokenImageUrl={[Function]}
              onSelectedImageChange={[Function]}
              productId={1}
              url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274"
            />,
            <ProductPreviewImageItem
              active={false}
              onBrokenImageUrl={[Function]}
              onSelectedImageChange={[Function]}
              productId={1}
              url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138"
            />,
            <ProductPreviewImageItem
              active={false}
              onBrokenImageUrl={[Function]}
              onSelectedImageChange={[Function]}
              productId={1}
              url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
            "nodeType": "function",
            "props": Object {
              "active": true,
              "onBrokenImageUrl": [Function],
              "onSelectedImageChange": [Function],
              "productId": 1,
              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
            "nodeType": "function",
            "props": Object {
              "active": false,
              "onBrokenImageUrl": [Function],
              "onSelectedImageChange": [Function],
              "productId": 1,
              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
            "nodeType": "function",
            "props": Object {
              "active": false,
              "onBrokenImageUrl": [Function],
              "onSelectedImageChange": [Function],
              "productId": 1,
              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
            "nodeType": "function",
            "props": Object {
              "active": false,
              "onBrokenImageUrl": [Function],
              "onSelectedImageChange": [Function],
              "productId": 1,
              "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Image
            alt="Produto"
            height={200}
            onBrokenImageUrl={[Function]}
            src="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
            width={200}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "Produto",
            "height": 200,
            "onBrokenImageUrl": [Function],
            "src": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
            "width": 200,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  justify-content: space-around;
  align-items: center;

  > div:first-child {
    display: flex;
    flex-direction: column;
  }
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div>
            <ProductPreviewImageItem
              active={true}
              onBrokenImageUrl={[Function]}
              onSelectedImageChange={[Function]}
              productId={1}
              url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
            />
            <ProductPreviewImageItem
              active={false}
              onBrokenImageUrl={[Function]}
              onSelectedImageChange={[Function]}
              productId={1}
              url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274"
            />
            <ProductPreviewImageItem
              active={false}
              onBrokenImageUrl={[Function]}
              onSelectedImageChange={[Function]}
              productId={1}
              url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138"
            />
            <ProductPreviewImageItem
              active={false}
              onBrokenImageUrl={[Function]}
              onSelectedImageChange={[Function]}
              productId={1}
              url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730"
            />
          </div>,
          <div>
            <Image
              alt="Produto"
              height={200}
              onBrokenImageUrl={[Function]}
              src="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
              width={200}
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ProductPreviewImageItem
                active={true}
                onBrokenImageUrl={[Function]}
                onSelectedImageChange={[Function]}
                productId={1}
                url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
              />,
              <ProductPreviewImageItem
                active={false}
                onBrokenImageUrl={[Function]}
                onSelectedImageChange={[Function]}
                productId={1}
                url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274"
              />,
              <ProductPreviewImageItem
                active={false}
                onBrokenImageUrl={[Function]}
                onSelectedImageChange={[Function]}
                productId={1}
                url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138"
              />,
              <ProductPreviewImageItem
                active={false}
                onBrokenImageUrl={[Function]}
                onSelectedImageChange={[Function]}
                productId={1}
                url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
              "nodeType": "function",
              "props": Object {
                "active": true,
                "onBrokenImageUrl": [Function],
                "onSelectedImageChange": [Function],
                "productId": 1,
                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
              "nodeType": "function",
              "props": Object {
                "active": false,
                "onBrokenImageUrl": [Function],
                "onSelectedImageChange": [Function],
                "productId": 1,
                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
              "nodeType": "function",
              "props": Object {
                "active": false,
                "onBrokenImageUrl": [Function],
                "onSelectedImageChange": [Function],
                "productId": 1,
                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
              "nodeType": "function",
              "props": Object {
                "active": false,
                "onBrokenImageUrl": [Function],
                "onSelectedImageChange": [Function],
                "productId": 1,
                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Image
              alt="Produto"
              height={200}
              onBrokenImageUrl={[Function]}
              src="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
              width={200}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": "Produto",
              "height": 200,
              "onBrokenImageUrl": [Function],
              "src": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
              "width": 200,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  justify-content: space-around;
  align-items: center;

  > div:first-child {
    display: flex;
    flex-direction: column;
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
